pkg/flare/archive.go

```diff
@@ -117,11 +123,39 @@ func CreatePerformanceProfile(pprofURL string, profileDuration int) (*Profile, e
 		return nil, err
 	}
 
-	return &Profile{
+	p := &Profile{
 		FirstHeapProfile:  firstHeapProf,
 		SecondHeapProfile: secondHeapProf,
 		CPUProfile:        cpuProf,
-	}, nil
+	}
+
+	if k := "apm_config.enabled"; config.Datadog.IsSet(k) && !config.Datadog.GetBool(k) {
+		return p, nil
+	}
+
+	apmPort := 8126
+	if v := config.Datadog.GetInt("apm_config.receiver_port"); v != 0 {
+		apmPort = v
+	}
+	tracePath := fmt.Sprintf("http://127.0.0.1:%d/debug/pprof", apmPort)
```

Comment:
I think if we refactor so that the function parameter `pprofURL` doesn't use the `expvar` port by default, we can use it both here and above for the Core Agent profile.

Reply:
I think you're right overall about separating the two type of profiles. It might be cleaner. I will look into it and put more work in to make sure it's nice.